public/src/managers/ServerProvider.ts

```diff
@@ -375,23 +374,20 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
 
     public async updateSubmission(courseID: number, submissionID: number, approve: boolean): Promise<boolean> {
         const result = await this.grpcHelper.updateSubmission(courseID, submissionID, approve);
-        if (result.status.getCode() !== 0) {
-            return false;
-        }
-        return true;
+        return this.responseCodeSuccess(result);
```

Comment:
I'm not sure if the `result.data` field is non-null for this case, when response code is success, but probably.

Reply:
So I think a rewritten method can still be used here.

Reply:
The gRPC method here returns `Void`, should we care whether it is here or not? Status code will be 0 if the request returned without errors and we check exactly that.

Reply:
This is fine to just check the status code.